play.pokemonshowdown.com/js/client.js

```diff
@@ -2555,6 +2555,21 @@ function toId() {
 			this.$el.html('<form><p style="white-space:pre-wrap;word-wrap:break-word">' + (data.htmlMessage || BattleLog.parseMessage(data.message)) + '</p><p class="buttonbar">' + (data.buttons || '<button type="button" name="close" class="button autofocus"><strong>OK</strong></button>') + '</p></form>').css('max-width', data.maxWidth || 480);
 		},
 
+		copyText: function (value, target) {
+			var dummyInput = document.createElement("input");
+			// This is a hack. You can only "select" an input field.
+			//  The trick is to create a short lived input element and destroy it after a copy.
+			// (stolen from the replay code, obviously --mia)
+			dummyInput.id = "dummyInput";
+			dummyInput.value = value || target.value || target.href || "";
+			dummyInput.style.position = 'absolute';
+			target.appendChild(dummyInput);
+			dummyInput.select();
+			document.execCommand("copy");
+			target.removeChild(dummyInput);
+			$(target).text('Copied!');
```

Comment:
I will ask you to make a change here though. The best way to handle this is going to be calling the existing method rather than copy/pasting the code. This is easier to maintain because if any changes need to be made to the code for copying text, it only needs to be changed in one place. This can be done with this:

```suggestion
			app.curRoom.copyText(value, target);
```

Reply:
Appreciate the review, can get this change done tomorrow AM!